src/api/cpp/cvc5.cpp

```diff
@@ -2721,6 +2752,414 @@ std::vector<Node> Term::termVectorToNodes(const std::vector<Term>& terms)
+  return d_node->getKind() == cvc5::Kind::CONST_FLOATINGPOINT
+         && d_node->getConst<FloatingPoint>().isNaN();
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+bool Term::isFloatingPoint() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  //////// all checks before this line
+  return d_node->getKind() == cvc5::Kind::CONST_FLOATINGPOINT;
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+std::tuple<std::uint32_t, std::uint32_t, Term> Term::getFloatingPoint() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  CVC5_API_CHECK(d_node->getKind() == cvc5::Kind::CONST_FLOATINGPOINT)
+      << "Term should be a FloatingPoint when calling getFloatingPoint()";
+  //////// all checks before this line
+  const auto& fp = d_node->getConst<FloatingPoint>();
+  return std::make_tuple(fp.getSize().exponentWidth(),
+                         fp.getSize().significandWidth(),
+                         d_solver->mkValHelper<BitVector>(fp.pack()));
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+
+bool Term::isSet() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  //////// all checks before this line
+  return d_node->getType().isSet();
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+
+void Term::collectSet(std::set<Term>& set, const cvc5::Node& node, const Solver* slv)
+{
+  switch (node.getKind())
+  {
+    case cvc5::Kind::EMPTYSET: break;
+    case cvc5::Kind::SINGLETON:
+      set.emplace(Term(slv, node[0])); break;
+    case cvc5::Kind::UNION:
+    {
+      for (const auto& sub : node)
+      {
+        collectSet(set, sub, slv);
+      }
+      break;
+    }
+    default: break;
+  }
+}
+
+std::set<Term> Term::getSet() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  CVC5_API_CHECK(d_node->getType().isSet())
+      << "Term should be a Set when calling getSet()";
+  //////// all checks before this line
+  std::set<Term> res;
+  Term::collectSet(res, *d_node, d_solver);
+  return res;
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+
+bool Term::isChar() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  //////// all checks before this line
+  return d_node->getKind() == cvc5::Kind::CONST_STRING
+         && d_node->getConst<String>().getVec().size() == 1;
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+std::string Term::getChar() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  CVC5_API_CHECK_NOT_NULL;
+  CVC5_API_CHECK(d_node->getKind() == cvc5::Kind::CONST_STRING
+                 && d_node->getConst<String>().getVec().size() == 1)
+      << "Term should be a Char when calling getChar()";
+  //////// all checks before this line
+  unsigned codePoint = d_node->getConst<String>().getVec()[0];
+  const char* digits = "0123456789ABCDEF";
+  std::string res;
+  for (; codePoint > 0; codePoint /= 16)
+  {
+    res += digits[codePoint % 16];
+  }
+  std::reverse(res.begin(), res.end());
+  return res;
+  ////////
+  CVC5_API_TRY_CATCH_END;
+}
+bool Term::isSequence() const
```

Comment:
As discussed offline, we should change this to `isConstSequence` which holds only when `getKind()==CONST_SEQUENCE`.